teps/0094-configuring-resources-at-runtime.md

```diff
@@ -52,12 +61,495 @@ While this may be a valuable feature, it should be considered in a separate prop
+(e.g. the example used in [Merging resource requirements](#merging-resource-requirements)).
+
+## Design Evaluation
+### Reusability
+This proposal increases reusability of `Task`s and `Pipeline`s by allowing
+environment-specific execution requirements to be updated at runtime.
+
+### Simplicity
+The proposed solution contains the minimum number of features that meet the specified
+requirements, compared to the listed alternatives. 
+
+### Flexibility
+This proposal increases Tekton's flexibility by giving users more options to modify `Task`s.
+There isn't a clear strategy for implementing this functionality via a plugin system.
+
+### Conformance
+Tekton aims to minimize Kubernetes-specific features in its API.
+However, the usage of `ResourceRequirements` is necessary for this feature, as a result of the
+decision to directly embed `Container` in the `Task` API.
+
+Container resource requirements are required for
+[Knative Serving conformance](https://github.com/knative/specs/blob/main/specs/serving/knative-api-specification-1.0.md#container)
+but not for
+[Tekton pipelines conformance](https://github.com/tektoncd/pipeline/blob/main/docs/api-spec.md#step).
+Therefore, `StepResources` and `SidecarResources` should also not be required for Tekton conformance.
+
+## Drawbacks
+
+In an ideal world, `Task`s would not contain fields that are tied to runtime requirements.
+`Task`s might be more reusable if `Step` and `Sidecar` were fully Tekton-owned,
+instead of having the `Container` API embedded.
+However, this would be a major API change at this point,
+for little gain compared to the proposed solution.
+
+## Alternatives
+
+### Allow `TaskRun`s to patch arbitrary fields of `Task`s
+
+Introduce new syntax to `TaskRunSpec` and `PipelineTaskRunSpec` to allow these structs
+to override any `Task` field, via a "path" key and a value.
+
+#### Example `Task` and `TaskRun`
+task.yaml:
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: image-build-task
+spec:
+  steps:
+    - name: build
+      image: gcr.io/kaniko-project/executor:latest
+      command:
+        - /kaniko/executor
+```
+
+taskrun.yaml:
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: image-build-taskrun
+spec:
+  taskRef:
+    name: image-build-task
+  patches:
+    - path: taskRef.steps[0].resources.requests.memory
+      value: 1Gi
+```
+This solution is not the proposed solution because it does not align
+with the design principle "Tekton should contain only the bare minimum and
+simplest features needed to meet the largest number of CI/CD use cases."
+Additional pros and cons are as follows:
+
+Pros:
+- Allows resource requirements to be specified for each `Step` and `Sidecar` at runtime.
+- Increases reusability of `Task`s by allowing catalog `Task`s to be modified.
+- Replacement of any other `Task` field comes for free, although this is a non-goal.
+- Syntax similarity to existing tools such as `kustomize`.
```

Comment:
if we went this route (which i hope we don't XD) we'd probably want to use JSONPath since we're settling on that syntax for variable replacement as well